index.js

```diff
@@ -206,6 +178,46 @@ function prettyFactory (options) {
     }
 
     return line
+
+    function createLineMetadata () {
```

Comment:
Does this function need to be defined as an inner function of `prettyFactory`? Or can we put it in `lib/utils`?

Reply:
We could put it in `lib/utils` but would have to pass many parameters like `log`, `colorizer`, `levelKey` etc. Think i would prefer to leave it in index.js. What do you think about the naming?

Reply:
My concern is legibility and maintainability. The inlined function is easier to read than the nested conditional version, but it is still a lot of code deeply nested within another function.

I think the function name is fine.

Reply:
@timlohse1104 I think you'll find it much easier to accommodate the ask now that #453 has been merged. Unfortunately, it'll be easier to start with a fresh PR. Rebasing this one may be too difficult.

If you have any questions, don't hesitate to ask.